portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -239,8 +241,22 @@ public DLFileEntry addFileEntry(
 			PortalUtil.getCurrentAndAncestorSiteGroupIds(groupId), folderId,
 			fileEntryTypeId);
 
+		String actualExtension = null;
+
+		if (inputStream instanceof ByteArrayFileInputStream) {
+			ByteArrayFileInputStream inputByteArrayFileInputStream =
+				(ByteArrayFileInputStream)inputStream;
+
+			File uploadedFile = inputByteArrayFileInputStream.getFile();
+
+			String actualName = uploadedFile.getName();
```

Comment:
If a variable is used only once (and there's no weird chaining involved) you can inline it.

Reply:
Ok, I've gone too far then